### DISCUSSION TITLE
Support for NASL templates

### DISCUSSION THREAD

**Discussion by david-colombo** [projectdiscovery/nuclei#673] [Ideas]
Hey folks :D

First of all, Nuclei is great and I appreciate all of the work put into it!
I can already see how it will possibly replace OpenVAS in my environment one day.
The only major problem I have with Nuclei is the very limited number of templates currently available.

So, from working with OpenVAS/GVM I have over 60k (Nessus Attack Scripting Language) NASL files over here and I've been thinking "what if Nuclei would support them". This would increase the available templates from a few hundred to tens of thousands instantly and immediately make Nuclei one of the best vulnerability scanner out there.

Did you ever think about something like this or do you rule it out completely?
I'd love to get all your opinions on this!

**Comment by ehsandeep** [comment]
Hey @david-colombo, 

we know this sounds exciting and discussed few times in the past, but practically not sure if it's possible as we have to write another complete engine that understands NASL format.
